extensions/pyRevitTools.extension/pyRevit.tab/Project.panel/ptools.stack/Family.pulldown/Family QuickCheck.pushbutton/FamilyQuickcheck_script.py

```diff
@@ -6,37 +6,66 @@
 """
 #pylint: disable=invalid-name,import-error,superfluous-parens,broad-except
 
-from pyrevit import revit, DB
-from pyrevit import script
+from pyrevit import revit, script, DB, forms
+from pyrevit.revit import FailureSwallowerContext
 
 
 __author__ = 'Frederic Beaupere\n{{author}}'
 
-
 logger = script.get_logger()
 output = script.get_output()
 
+checked_families = 0
+count_errors = 0
+count_warnings = 0
+cancelled = False
+
+def update_output_title():
+    output.set_title("Family QuickCheck (Err:{} Warn:{})"
+                     .format(count_errors, count_warnings))
+
+
 all_families = DB.FilteredElementCollector(revit.doc)\
                  .OfClass(DB.Family)\
                  .ToElements()
 
 editable_families = [x for x in all_families if x.IsEditable]
 total_count = len(editable_families)
-checked_families = 0
 
 output.reset_progress()
 
-for fam in editable_families:
-    fam_name = revit.query.get_name(fam)
-    logger.debug("attempt to open family: %s", fam_name)
+with FailureSwallowerContext(force=True) as swallower:
     try:
```

Comment:
I find it is important to wrap it into try just bellow `with`. To be sure that the event will be unregistered in the end